modal-go/function.go

```diff
@@ -0,0 +1,122 @@
+package modal
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	pickle "github.com/kisielk/og-rek"
+	proto "github.com/modal-labs/libmodal/modal-go/proto/modal_proto"
+)
+
+func timeNow() float64 {
+	return float64(time.Now().UnixNano()) / 1e9
+}
+
+// function references a deployed Modal Function.
+type Function struct {
+	FunctionId string
+	ctx        context.Context
+}
+
+// FunctionLookup looks up an existing Function.
+func FunctionLookup(ctx context.Context, appName string, name string, options LookupOptions) (*Function, error) {
+	ctx = clientContext(ctx)
+
+	resp, err := client.FunctionGet(ctx, proto.FunctionGetRequest_builder{
+		AppName:         appName,
+		ObjectTag:       name,
+		Namespace:       proto.DeploymentNamespace_DEPLOYMENT_NAMESPACE_WORKSPACE,
+		EnvironmentName: environmentName(options.Environment),
```

Comment:
I wrote an `environmentName()` helper which handles the default environment in your modal.toml profile and `MODAL_ENVIRONMENT` variable